src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,290 @@
-import { ResultList } from '../ResultList/ResultList';
-import { IResultListOptions } from '../ResultList/ResultListOptions';
 import { IComponentBindings } from '../Base/ComponentBindings';
 import { exportGlobally } from '../../GlobalExports';
-import { ComponentOptions, OmniboxEvents, Initialization } from '../../Core';
+import { ComponentOptions, OmniboxEvents, Initialization, $$, Component, Assert, QueryUtils, Dom } from '../../Core';
 import { IQuerySuggestSelection } from '../../events/OmniboxEvents';
+import { IQueryResults } from '../../rest/QueryResults';
+import 'styling/_QuerySuggestPreview';
+import { ResultListRenderer } from '../ResultList/ResultListRenderer';
+import { IInitializationParameters, IInitResult } from '../Base/Initialization';
+import { IResultListOptions } from '../ResultList/ResultListOptions';
+import { Template } from '../Templates/Template';
+import { TemplateComponentOptions } from '../Base/TemplateComponentOptions';
+import { IQueryResult } from '../../rest/QueryResult';
+import { pluck, sortBy, map } from 'underscore';
+import { ResultListTableRenderer } from '../ResultList/ResultListTableRenderer';
 
-export interface IQuerySuggestPreview extends IResultListOptions {
+export interface IQuerySuggestPreview {
   numberOfPreviewResults?: number;
-  width?: string;
+  previewWidth?: number;
+  suggestionWidth?: string;
+  resultTemplate?: Template;
+  headerText?: string;
+  hoverTime?: number;
 }
 
-export class QuerySuggestPreview extends ResultList implements IComponentBindings {
+/**
```

Comment:
The `QuerySuggestPreview` component allows you to preview the top query result items of hovered query suggestions.

This component requires a working [`querySuggest`]{@link Omnibox.options.enableQuerySuggestAddon} component in the corresponding search box, and you need to provide a [result template](https://developers.coveo.com/x/aIGfAQ) for the preview.

**Example**
```
<div class="CoveoQuerySuggestPreview">
  <script class="result-template" type="text/x-underscore">
    <span>
      <a class='CoveoResultLink'></a>
    </span>
  </script>
</div>
```